bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1175,7 +1176,30 @@ protected synchronized void removeThingHandlerFactory(ThingHandlerFactory thingH
     @Override
     public void onReadyMarkerAdded(ReadyMarker readyMarker) {
         startLevelSetterJob = scheduler.scheduleWithFixedDelay(() -> {
-            if (things.values().stream().anyMatch(t -> !ThingHandlerHelper.isHandlerInitialized(t) && t.isEnabled())) {
+            // check if all things are ready
+            for (Thing thing : things.values()) {
+                if (!thing.isEnabled() || ThingHandlerHelper.isHandlerInitialized(thing)) {
+                    continue;
+                }
+
+                int bridgeNestingLevel = 0;
+                boolean bridgeDisabled = false;
+                Bridge bridge = getBridge(thing.getBridgeUID());
+                while (bridge != null) {
+                    if (!bridge.isEnabled()) {
+                        bridgeDisabled = true;
+                        break;
+                    }
+                    bridge = getBridge(bridge.getBridgeUID());
+                    if (bridgeNestingLevel++ > MAX_BRIDGE_NESTING) {
```

Comment:
Why do you limit the nesting here? I agree that there is probably no use-case for 100 levels of bridge-nesting, but what would be the problem here? Potential stack overflow if nesting is too deep?

Reply:
avoid an endless loop due to circular reference